packages/unlock-js/src/PublicLock/v7/setKeyManagerOf.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Change lock manager for a specific key
+ * @param {string} lockAddress address of the lock
+ * @param {string} managerAddress address of the user
+ */
+
+export default async function (
+  { lockAddress, managerAddress, tokenId },
+  transactionOptions = {},
+  callback
+) {
+  const lockContract = await this.getLockContract(lockAddress)
+  const transactionPromise = lockContract.setKeyManagerOf(
+    tokenId,
+    managerAddress
+  )
+
+  const hash = await this._handleMethodCall(transactionPromise)
+
+  if (callback) {
+    callback(null, hash)
+  }
+
+  await this.provider.waitForTransaction(hash)
+
+  return null
```

Comment:
It could be interesting to see if we want to return something more specific here?

Reply:
what we can do here eventually is start to use this approach 

https://github.com/unlock-protocol/unlock/pull/11327#discussion_r1139155491